packages/grafana-ui/src/components/Logs/LogRowContext.tsx

```diff
@@ -197,18 +197,17 @@ export const LogRowContext: React.FunctionComponent<LogRowContextProps> = ({
   onLoadMoreContext,
   hasMoreContextRows,
 }) => {
-  const handleEscKeyDown = (e: KeyboardEvent): void => {
-    if (e.keyCode === 27) {
-      onOutsideClick();
-    }
-  };
-
   useEffect(() => {
+    const handleEscKeyDown = (e: KeyboardEvent): void => {
+      if (e.keyCode === 27) {
+        onOutsideClick();
+      }
+    };
     document.addEventListener('keydown', handleEscKeyDown, false);
     return () => {
       document.removeEventListener('keydown', handleEscKeyDown, false);
     };
-  }, []);
+  }, [onOutsideClick]);
```

Comment:
nit: Is this a deliberate change?

Reply:
It is a deliberate change. We should run this if onOutsideClick is changed for sure.